src/cryptoManager.ts

```diff
@@ -30,26 +36,36 @@ export class CryptoManager {
     }
   }
 
-  private async computeHash(keyFilePath: string): Promise<string> {
-    this.logger.debug('computing hash key');
-    const secret = await fsp.readFile(keyFilePath, { encoding: 'binary' });
-    const hash = crypto.createHash('sha256');
+  private async computeHash(filePath: string): Promise<Buffer> {
+    const secret = await fsp.readFile(filePath, { encoding: 'binary' });
+    const hash = crypto.createHash('sha512');
     hash.update(String(secret));
-    const hashKey = hash.digest('base64');
+    const hashKey = hash.digest();
     return hashKey;
   }
 
-  private encryptHash(fileHash: string): IEncryptedHash {
+  private encryptHash(fileHash: Buffer): IEncryptedHash {
     const ivSize = 16;
-    this.logger.debug('encrypting hash');
     const iv = Buffer.allocUnsafe(ivSize);
-    const cipher = crypto.createCipheriv('aes-256-cfb', Buffer.from(fileHash, 'base64'), iv);
+    const cipher = crypto.createCipheriv('aes-256-cfb', this.key, iv);
```

Comment:
move to config (no need for environment variable)

Reply:
fixed